### PR TITLE
Move from sequential to concurrent execution model; remove scheduler interface and refactor

### PR DESCRIPTION
**Pull request by nrc**
This PR removes the sequential scheduler and lets each request execute on its own green thread. I also removed the Scheduler interface, moved code around to avoid import cycles and to make a cleaner package layout and simplified several parts.

PTAL @Connor1996

### CODE REVIEW

File: kv/tikv/storage/server.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+		reader.Close()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if latches != nil {
+		// The command will write to the DB.
+		for {
+			wg := server.Latches.AcquireLatches(latches)
```
Comment (Connor1996) [inline kv/tikv/storage/server.go]: Add a function `WaitLatches` for `Latches`?

File: kv/tikv/storage/server.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+func (server *Server) KvResolveLock(ctx context.Context, req *kvrpcpb.ResolveLockRequest) (*kvrpcpb.ResolveLockResponse, error) {
+	cmd := NewResolveLock(req)
+	resp, err := server.Run(&cmd)
+	return resp.(*kvrpcpb.ResolveLockResponse), err
+}
+
+// Raw API.
+func (server *Server) RawGet(ctx context.Context, req *kvrpcpb.RawGetRequest) (*kvrpcpb.RawGetResponse, error) {
+	cmd := NewRawGet(req)
```
Comment (Connor1996) [inline kv/tikv/storage/server.go]: make raw API back to https://github.com/pingcap-incubator/tinykv/blob/6a6122f5842d7c8b28e66b5f4cfcb82e5181b0fb/kv/tikv/server.go#L134
and remove the related commands

When implementing LAB1, we don't want to involve the term `Latch` and `MvccTxn`

File: kv/tikv/storage/server.go [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+}
+
+func NewServer(innerServer InnerServer) *Server {
+	return &Server{
+		innerServer: innerServer,
+		Latches:     latches.NewLatches(),
+	}
+}
+
+func (server *Server) Run(cmd Command) (interface{}, error) {
```
Comment (Connor1996) [inline kv/tikv/storage/server.go]: I'm not sure it is necessary to extract the command the logic still after the scheduler is removed.
Writing individual logic for every transaction API seems is simpler with just a little boilerplate code, see
https://github.com/ngaut/unistore/blob/master/tikv/server.go#L287
Though both look fine for me.

File: kv/tikv/storage/server.go (hunk 1 above)
Comment (nrc) [reply kv/tikv/storage/server.go]: Shall do

**Comment by nrc** [comment]
I prefer to discuss the design as a whole and then we can make a follow-up PR, if that is OK with you?

**Comment by Connor1996** [comment]
@nrc of course, maybe hold a meeting to discuss it? When are you available?

**Comment by nrc** [comment]
I've added `WaitForLatches` function. I'll leave the rest for another PR. PTAL @Connor1996

**Pull request merged**